### PR TITLE
Add attributes to ebs volumes resource

### PR DESCRIPTION
**Pull request by jnikles1**
Signed-off-by: Justin Nikles <justin.nikles@sap.com>

### Description

Added attributes to the ebs volumes resource so that additional attributes other than the volume_id can be accessed

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [x] Documentation provided or updated for resources
- [x] All Integration Tests pass 
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_ebs_volumes.rb [hunk 1]
```diff
@@ -33,7 +48,24 @@ def fetch_data
       return [] if !@api_response || @api_response.empty?
```
Comment (soumyo13) [inline libraries/aws_ebs_volumes.rb:48]:
```suggestion
      return volume_rows if !@api_response || @api_response.empty?
```

**Review by soumyo13 — CHANGES_REQUESTED** [review]
Please update the documentation, integration tests, and also unit tests. Please add the parameter max_results in the plural resource.

**Comment by jnikles1** [comment]
@soumyo13 where would you like me to add the max_results parameter? Looking at this resource as an example but I don't see any max results parameter https://github.com/inspec/inspec-aws/blob/main/libraries/aws_efs_file_systems.rb

**Comment by jnikles1** [comment]
@soumyo13 do you have any documentation on creating these integration and unit tests?

**Comment by soumyo13** [comment]
> @soumyo13 where would you like me to add the max_results parameter? Looking at this resource as an example but I don't see any max results parameter https://github.com/inspec/inspec-aws/blob/main/libraries/aws_efs_file_systems.rb

https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/EC2/Client.html#describe_volumes-instance_method
Please see the above link. And also see the response.

Resp:
resp = client.describe_volumes({
  filters: [
    {
      name: "String",
      values: ["String"],
    },
  ],
  volume_ids: ["VolumeId"],
  dry_run: false,
  **max_results: 1,**
  next_token: "String",
})

**Comment by soumyo13** [comment]
> @soumyo13 do you have any documentation on creating these integration and unit tests?

Please check the below links for the following:
For documentation: https://github.com/inspec/inspec-aws/blob/main/docs/resources/aws_ebs_volumes.md
As you have added other parameters, you have to update the docs with the parameters that you have added. As of now there is only one parameter that is added.

For unit tests: You have to write the unit test as you have added the parameters you have added in the plural resource.

For integration tests: Please add the integration tests which will test the parameters that you have added in the plural resource.

**Comment by jnikles1** [comment]
> > @soumyo13 where would you like me to add the max_results parameter? Looking at this resource as an example but I don't see any max results parameter https://github.com/inspec/inspec-aws/blob/main/libraries/aws_efs_file_systems.rb
> 
> https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/EC2/Client.html#describe_volumes-instance_method Please see the above link. And also see the response.
> 
> Resp: resp = client.describe_volumes({ filters: [ { name: "String", values: ["String"], }, ], volume_ids: ["VolumeId"], dry_run: false, **max_results: 1,** next_token: "String", })

Hi @soumyo13 - still not understanding as I don't see this parameter in other resources. Can you point me to an example so that I can see an example implementation?

**Comment by soumyo13** [comment]
Can we have a call tomorrow?

**Comment by jnikles1** [comment]
Yes that would be great

**Comment by soumyo13** [comment]
@jnikles1 Any update on this?

**Comment by jnikles1** [comment]
@soumyo13 - apologies, was away for the holidays. The work is almost done and should be pushed today

File: test/unit/resources/aws_ebs_volumes_test.rb [hunk 2]
```diff
@@ -20,11 +20,33 @@ def test_volumes_non_existing_for_empty_response
 class AwsEbsVolumesHappyPathTest < Minitest::Test
 
   def setup
+    attachment_state = %w(attaching attached detaching)
+    encrypted = [true, false]
+    volume_id = "ebs-12345"
```
Comment (Nirbhay1997) [inline test/unit/resources/aws_ebs_volumes_test.rb:25]: Do we need this line  ??

Comment (Nirbhay1997) [inline test/unit/resources/aws_ebs_volumes_test.rb:25]:
```suggestion
  
```

Comment (Nirbhay1997) [inline test/unit/resources/aws_ebs_volumes_test.rb:25]:
```suggestion
```

**Review by Nirbhay1997 — APPROVED** [review]
LGTM   :100:

**Review by dkumaras — APPROVED** [review]

**Pull request merged**